### ISSUE TITLE
react/prefer-stateless-function with PureComponent.

### ISSUE THREAD

**Issue by ghost** [yannickcr/eslint-plugin-react#1773]
I'm using:

`"react/prefer-stateless-function": [1, { "ignorePureComponents": true }] `

but I'm also using this new `Query` component by `react-apollo`:

```js
import React, { PureComponent } from "react";
import gql from "graphql-tag";
import { Query } from "react-apollo";

const GET_DOGS = gql`
  {
    dogs {
      id
      breed
    }
  }
`;

class DogsLove extends PureComponent {
    render() {
        return (
            <Query query={GET_DOGS}>
                {({ loading, error, data }) => {
                if (loading) return "Loading...";
                if (error) return `Error! ${error.message}`;

                return (
                    <select name="dog" onChange={onDogSelected}>
                    {data.dogs.map(dog => (
                        <option key={dog.id} value={dog.breed}>
                        {dog.breed}
                        </option>
                    ))}
                    </select>
                );
                }}
            </Query>
        )
    }
}
```

So, **why this is a warning**? (Or an error if I disable `ignorePureComponents`?)

I just need PureComponent because I don't need to re-render `DogsLove` component every time I switch Route. I need it to render just one time, the first time. Then `Query` component is making an observable and nothing more.

Where am I wrong?

**Comment by johnunclesam** [comment]
Scenario in here: https://github.com/apollographql/react-apollo/issues/1938

**Comment by ljharb** [comment]
Can you paste the actual warning message (and denote the line it points to)?

**Comment by ljharb** [comment]
Separately, if you truly only want to render once, you’d want a normal class component with a shouldComponentUpdate that returns false.

**Comment by johnunclesam** [comment]
> Can you paste the actual warning message (and denote the line it points to)?

All the component is in warning:

`Component should be written as a pure function (react/prefer-stateless-function)`

**Comment by ljharb** [comment]
@johnunclesam what if you `extends React.PureComponent` (instead of bringing it in as a named import)?

**Comment by D-e-e-m-o** [comment]
In `lib/rules/prefer-stateless-function.js` line 374,I find this code
```JavaScript
if (list[component].hasSCU && list[component].usePropsOrContext) {
    continue;
}
```
But,I find that `list[component].usePropsOrContext === undefined` ,so that the next code isn't skiped.
So what's the usage of `list[component].usePropsOrContext`?Can I delete  it? @yannickcr

**Comment by JavierLaguna** [comment]
I have the same problem. ignorePureComponents: true doesn't seem to do anything.

Doesn't matter if I extend PureComponent or React.PureComponent.

**Comment by nickbouton** [comment]
Same problem here as well. If you add props or anything beyond a simple `return` to `render`, though, it stops complaining.

**Comment by FlaviooLima** [comment]
This bug need some attention :)

**Comment by golopot** [comment]
Fixed by #2238?

https://github.com/yannickcr/eslint-plugin-react/blob/6e4f43ef4739191a2157659de178d1cd9a28a42e/tests/lib/rules/prefer-stateless-function.js#L310-L321

**Comment by JavierLaguna** [comment]
The fix is merged on the current version v7.12.4 ?

**Comment by golopot** [comment]
No, the pr was merged 18 days ago while v7.12.4 is released 3 months ago.

**Comment by JavierLaguna** [comment]
> No, the pr was merged 18 days ago while v7.12.4 is released 3 months ago.

Thank you!
